figures/Fig1PCA.Rmd

```diff
@@ -123,19 +154,123 @@ as.mat<-mvscores%>%subset(specimenID%in%rownames(samp.data))%>%
   as.matrix()
 
-autoplot(prcomp(t(as.mat)),data=samp.data,colour="tumorType",shape='study')+ggtitle('LV-derived clustering')+theme_bw()+scale_color_manual(values = c("Cutaneous Neurofibroma"="#ca054d", 
+comps<-prcomp(t(as.mat))
+
+dist.vals<-as.dist(comps$x)%>%
+  as.matrix()%>%abs()%>%
+  as.data.frame()%>%
+  rownames_to_column('from')%>%
+  pivot_longer(cols=(-from),names_to='to')
+
+lv.tt.df<-dist.vals%>%
+  left_join(rename(tumType,from=sample,fromType=tumorType),by='from')%>%
+  left_join(rename(tumType,to=sample,toType=tumorType),by='to')%>%
+  filter(fromType==toType)%>%
+  rowwise()%>%
+  mutate(fromto=paste(sort(c(from,to)),collapse=','))%>%
+  select(fromto,value,fromType)%>%distinct()
+#  group_by(fromType)%>%summarize(meanDist=mean(value))
+
+#DT::datatable(lv.tt.df)
+
+lv.study.df<-dist.vals%>%
+  left_join(rename(tumType,from=sample,fromStudy=study),by='from')%>%
+  left_join(rename(tumType,to=sample,toStudy=study),by='to')%>%
+  filter(fromStudy==toStudy)%>%
+    rowwise()%>%
+  mutate(fromto=paste(sort(c(from,to)),collapse=','))%>%
+  select(fromto,value,fromStudy)%>%distinct()
+#  group_by(fromStudy)%>%summarize(meanDist=mean(value))
+
+#DT::datatable(lv.study.df)
+
+
+autoplot(comps,data=samp.data,colour="tumorType",shape='study')+ggtitle('LV-derived clustering')+theme_bw()+scale_color_manual(values = c("Cutaneous Neurofibroma"="#ca054d", 
                                "Plexiform Neurofibroma" = "#3b1c32",
                                "Neurofibroma" = "#a4d4b4",
                                "Malignant Peripheral Nerve Sheath Tumor" = "#ffcf9c"))
 
-#pdf('fig1panelB.pdf',width=8,height=5)
-autoplot(prcomp(t(as.mat)),data=samp.data,colour="tumorType",shape='study')+ggtitle('LV-derived clustering')+theme_bw()+
-  scale_color_manual(values = c("Cutaneous Neurofibroma"="#ca054d", 
+
+ggsave('fig1panelB.pdf',useDingbats=FALSE)
+#dev.off()
+
+
+```
+
+## Summary stats
+
+We can summarize the distances between points.
+
+```{r summary stats}
+
+tt.tab<-gene.tt.df%>%
+  select(fromto,value)%>%
+    rename(genes=value)%>%
+    left_join(lv.tt.df,by='fromto')%>%
+    rename(LVs=value,tumorType=fromType)%>%
+  pivot_longer(cols=c(genes,LVs),names_to='Values Clustered')
+
+study.tab<-gene.study.df%>%
+  select(fromto,value)%>%
+    rename(genes=value)%>%
+    left_join(lv.study.df,by='fromto')%>%
+    rename(LVs=value,study=fromStudy)%>%
+  pivot_longer(cols=c(genes,LVs),names_to='Values Clustered')
+
+full.tab<-rbind(tt.tab%>%select(fromto,`Values Clustered`,`Pairwise Distance`=value,Group=tumorType)%>%mutate(`Distance Measured`='Within Tumor Type'),study.tab%>%select(fromto,`Values Clustered`,`Pairwise Distance`=value,Group=study)%>%mutate(`Distance Measured`='Batch'))
+
+sigs<-full.tab%>%group_by(`Distance Measured`)%>%pivot_wider(names_from='Values Clustered',values_from='Pairwise Distance')%>%group_by(`Distance Measured`)%>%mutate(pval=t.test(genes,LVs)$p.value)%>%select(`Distance Measured`,pval)%>%distinct()
```

Comment:
Is this line doing a t-test for the paired distances? In that case we need to update the manuscript accordingly.

Reply:
Hmm, this is a good point, i wanted to not do a t-test.